src/txBaseMsgProcessor.h

```diff
@@ -0,0 +1,112 @@
+    // End of Tx/Certs processing Section
+
+    // Orphan Txes/Certs Tracker section
+    bool AddOrphanTx(const CTransactionBase& txObj, NodeId peer) EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+    void EraseOrphanTx(const uint256& hash)                      EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+    bool IsOrphan(const uint256& txBaseHash) const               EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+    const CTransactionBase* PickRandomOrphan()                   EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+    unsigned int LimitOrphanTxSize(unsigned int nMaxOrphans)     EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+    unsigned int countOrphans() const                            EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+
+    void EraseOrphansFor(NodeId peer)                            EXCLUSIVE_LOCKS_REQUIRED(cs_orphanTxes);
+    // End of Orphan Txes/Certs Tracker section
+
+    // Rejected Txes/Certs Tracker section
+    void SetupRejectionFilter(unsigned int nElements, double nFPRate) EXCLUSIVE_LOCKS_REQUIRED(cs_rejectedTxes);
+    bool HasBeenRejected(const uint256& txBaseHash) const             EXCLUSIVE_LOCKS_REQUIRED(cs_rejectedTxes);
+    void RefreshRejected(const uint256& currentTipHash)               EXCLUSIVE_LOCKS_REQUIRED(cs_rejectedTxes);
+    // End of Rejected Txes/Certs Tracker section
+
+private:
+    TxBaseMsgProcessor();
+    ~TxBaseMsgProcessor();
+
+    TxBaseMsgProcessor(const TxBaseMsgProcessor&) = delete;
+    TxBaseMsgProcessor& operator=(const TxBaseMsgProcessor&) = delete;
+
+    mutable CCriticalSection cs_orphanTxes;
+    struct COrphanTx {
+        std::shared_ptr<const CTransactionBase> tx;
+        NodeId fromPeer;
+    };
+    std::map<uint256, COrphanTx>          mapOrphanTransactions       GUARDED_BY(cs_orphanTxes);
+    std::map<uint256, std::set<uint256> > mapOrphanTransactionsByPrev GUARDED_BY(cs_orphanTxes);
+
+    /**
+     * Filter for transactions that were recently rejected by
+     * AcceptToMemoryPool. These are not rerequested until the chain tip
+     * changes, at which point the entire filter is reset. Protected by
+     * cs_main.
+     *
+     * Without this filter we'd be re-requesting txs from each of our peers,
+     * increasing bandwidth consumption considerably. For instance, with 100
+     * peers, half of which relay a tx we don't accept, that might be a 50x
+     * bandwidth increase. A flooding attacker attempting to roll-over the
+     * filter using minimum-sized, 60byte, transactions might manage to send
+     * 1000/sec if we have fast peers, so we pick 120,000 to give our peers a
+     * two minute window to send invs to us.
+     *
+     * Decreasing the false positive rate is fairly cheap, so we pick one in a
+     * million to make it highly unlikely for users to have issues with this
+     * filter.
+     *
+     * Memory used: 1.7MB
+     */
+    mutable CCriticalSection cs_rejectedTxes;
+    boost::scoped_ptr<CRollingBloomFilter> recentRejects  GUARDED_BY(cs_rejectedTxes);
```

Comment:
If there is no particular reason behind it, I would suggest to switch from `boost::scoped_ptr` to `std::unique_ptr` to reduce unnecessary dependency from Boost library.

```suggestion
    std::unique_ptr<CRollingBloomFilter> recentRejects  GUARDED_BY(cs_rejectedTxes);
```